pandas/tests/indexing/test_loc.py

```diff
@@ -2685,6 +2685,18 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
 
 
+def test_loc_setitem_multiindex_timestamp():
+    # GH#13831
+    vals = np.random.randn(8, 6)
+    idx = date_range("1/1/2000", periods=8)
+    cols = ["A", "B", "C", "D", "E", "F"]
+    df_mt = DataFrame(vals, index=idx, columns=cols)
+    df_mt.loc[df_mt.index[1], ("A", "B")] = np.nan
+    vals[1][0:2] = np.nan
+    res = DataFrame(vals, index=idx, columns=cols)
+    tm.assert_frame_equal(res, df_mt)
```

Comment:
Could you call result and expected appropriately? This would indicate better what is actually tested here, because this is not obvious to me

Reply:
Done